aten/src/ATen/native/cuda/AccumulateType.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+// Defines the accumulation type for a scalar type.
+// Example:
+//   using acc_scalar = cuda::acc_type<scalar>;
+
+#include <cuda.h>
+#include <cuda_fp16.h>
+
+namespace at { namespace native { namespace cuda {
+
+template <typename T>
+struct AccumulateType { using type = T; };
+
+template <>
+struct AccumulateType<half> { using type = float; };
```

Comment:
why are the int types not specified as they are in THC?  (I know you don't use them, but if someone else starts using them they'll screw things up).

Reply:
Good point. I'll add them.

Reply:
It's also nice that we can see the definitions all in one place instead of having to look through 10 files...